index.js

```diff
@@ -120,55 +120,57 @@ function checkForMatchInMessage(messageContent, matchingPattern) {
 }
 
 async function getMessageInfo(message) {
-    const result = await message;
+    const result = isPromise(message) ? await message : message;
     const messageDate = result.createdAt.toLocaleString();
     const messageAuthorId = result.author.id
-
     return {
         messageDate: messageDate,
         messageAuthorId: messageAuthorId
     }
 }
 
-async function checkPinsForMessageTracker(pins, generalChat) {
-    let messageTrackerExists = false; 
+async function getPins() {
+    const generalChat = getTargetChannel(process.env.GENERAL_CHAT_ID);
+    const pins = await generalChat.messages.fetchPinned();
+    return pins;
+}
 
+async function checkPinsForMessageTracker(pins) {
     for (const msg of pins.values()) {
         const messageTrackerMatch = checkForMatchInMessage(msg.content, MESSAGE_TRACKER_REGEX);
-        console.log(`Herre's messageTrackerMatch: ${messageTrackerMatch}`);
         if (messageTrackerMatch && messageTrackerMatch[0]) {
-            messageTrackerExists = true;
-            const messageTrackerId = msg.id;
-            console.log(`WE'RE IN`);
-            const messageTracker = await generalChat.messages.fetch(messageTrackerId);
-            console.log(`Here's messageTracker right after the await in the function: ${messageTracker}`);
-
-            return {    
-                messageTrackerExists: messageTrackerExists, 
-                messageTracker: messageTracker
-            };
+            MESSAGE_TRACKER_ID = msg.id;
+            return true;
         }
     };
+    return false;
+}
 
-    return {
-        messageTrackerExists: messageTrackerExists, 
-        messageTracker: null
-    };
+function checkForMessageTracker() {
+    if (MESSAGE_TRACKER_ID) {
+        return true;
+    }
+    return false;
+}
+
+function getPositionToAddReviewer(currentMessageTrackerContentByLineArray, review, reviewerMention) {
```

Comment:
I really like how you're fully using functions and modularizing all your code! Makes it really easy to follow throughout the file and it's a great sign of how quickly you're progressing as a developer. Even a lot of senior engineers forget this one and write giant 100 line functions haha

But let's kick it up a notch. I have a challenge for you: do you think we can simplify the `getPositionToAddReviewer`, `getPositionToAddReviewOne`, `getPositionToAddReviewTwo`? Note: this is one of those deceptively simple questions where you have anything from a 5th-grader level answer, college-student level answer, but PhD level answer too. I'm excited to see what your solution to this is going to be!